### ISSUE TITLE
Fix S3241 FP: Do not raise if the method is used as an argument

### ISSUE THREAD

**Issue by mary-georgiou-sonarsource** [SonarSource/sonar-dotnet#7429]
### Repro steps

```csharp
void Method(AnInterface foo)
{
    Func(0);
    foo?.DoSomething(Func);

    bool Func(int value) => true; // FP
}

public interface AnInterface
{
    void DoSomething(Func<int, bool> method);
}
```

### Expected behavior

Should not raise an issue, when the delegate of the method is used as an argument.

### Known workarounds

Use discard for the output.

### Related information

* C#/VB.NET Plugins version: 9.3

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1727](https://sonarsource.atlassian.net/browse/NET-1727)

[NET-1727]: https://sonarsource.atlassian.net/browse/NET-1727?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
